### PR TITLE
MRG : coreg for MGH images + cosmits

### LINKED ISSUE #279: pb coreg
I gave a try to Coregister using T1 MRs with:

```
from pypreprocess.coreg import Coregister
import nibabel as nib

source = nib.load('/Users/alex/mne_data/MNE-sample-data/subjects/sample/mri/T1.mgz')
target = nib.load('/Users/alex/mne_data/MNE-sample-data/subjects/fsaverage/mri/T1.mgz')

coreg = Coregister()
coreg.fit(source, target)
source_coreg = coreg.predict(source)
```

and I get

```
AttributeError: 'numpy.ndarray' object has no attribute 'get_data'
```

if I pass:

```
source = '/Users/alex/mne_data/MNE-sample-data/subjects/sample/mri/T1.mgz'
target = '/Users/alex/mne_data/MNE-sample-data/subjects/fsaverage/mri/T1.mgz'
```
I have a the same problem.

is it a version problem?

### PR DESCRIPTION
**Pull request by agramfort**
closes #279

but my attempt is not very  successful:

screen shot 2017-05-08 at 18 21 21

obtained with:

```
from pypreprocess.coreg import Coregister
import nibabel as nib

source = '/Users/alex/mne_data/MNE-sample-data/subjects/sample/mri/T1.mgz'
target = '/Users/alex/mne_data/MNE-sample-data/subjects/fsaverage/mri/T1.mgz'

coreg = Coregister()
coreg.fit(source, target)
source_coreg = coreg.transform(source)
source_coreg = source_coreg[0]
nib.save(source_coreg, 'T1_sample2fsaverage.nii')
```

### CODE REVIEW

File: pypreprocess/coreg.py [hunk 1]
```diff
@@ -358,7 +358,7 @@ def fit(self, target, source):
         self.params_ = np.array(self.params_init)
         for samp in self.sep:
             print("\r\nRunning Powell gradient-less local optimization "
-                  "(pyramidal level = %smm)...") % samp
+                  "(pyramidal level = %smm)..." % samp)
```
Comment (agramfort) [inline pypreprocess/coreg.py]: this was broken

File: pypreprocess/io_utils.py [hunk 2]
```diff
@@ -31,7 +31,8 @@ def is_niimg(img):
     if isinstance(img, (nibabel.Nifti1Image,
                         nibabel.Nifti1Pair,
                         nibabel.Spm2AnalyzeImage,
-                        nibabel.Spm99AnalyzeImage
+                        nibabel.Spm99AnalyzeImage,
+                        nibabel.MGHImage,
```
Comment (agramfort) [inline pypreprocess/io_utils.py]: this was the problem...

shall we use the nibabel base class? SpatialImage?

**Comment by bthirion** [comment]
Indeed... Just to check. In this type of case, I have no other suggestion than doing an approximate manual pre-registration...

**Comment by bthirion** [comment]
The changes look good otherwise, thx.

**Comment by agramfort** [comment]
what is my other automate option for this problem? thoughts? thanks for any
help and suggestion

**Comment by bthirion** [comment]
Try with ANTS...

**Comment by agramfort** [comment]
arfff Ok

I was hoping to avoid a painful dependency for MNE :'(

**Pull request merged**